### PR TITLE
Allow manual checkpoint sync without blobs

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

Since merging this PR, we don't need `--checkpoint-blobs`, even prior to Fulu:

- https://github.com/sigp/lighthouse/pull/8417

This PR removes the mandatory check for blobs prior to Fulu, enabling simpler manual checkpoint sync.

### CODE REVIEW

**Review by jimmygchen — COMMENTED** [review]
## Code Review

### ✅ Strengths

1. **Proper prerequisite**: Correctly builds on #8417 which enables backfill to re-import the checkpoint sync block
2. **Safety compliance**: No , , or unsafe array indexing
3. **Logic simplification**: Removes complex `is_before_fulu` check, making code more straightforward
4. **Clear documentation**: Updated comment explains the change well

### ⚠️ Recommendations

#### 1. Missing test coverage
The existing test at `beacon_node/beacon_chain/tests/store_tests.rs:2874` still provides blobs via `wss_blobs_opt.clone()`. There's no test covering the new code path where blobs are NOT provided and backfill must fetch them.

**Recommendation**: Add a test case that passes `None` for `anchor_blobs_bytes` to verify backfill correctly handles this scenario.

#### 2. Consider adding logging
When a block has blobs but the user doesn't provide them, there's no log message indicating that backfill will handle fetching them. This could aid debugging.

**Suggestion**: Add a log when blobs are omitted:
```rust
} else {
    if anchor_block.message().body().has_blobs() {
        info!("Checkpoint block has blobs/columns which will be fetched via backfill");
    }
    None
};
```

#### 3. Documentation updates
Since this changes user-facing behavior (users no longer need `--checkpoint-blobs`), verify these don't need updates:
- `book/src/advanced_checkpoint_sync.md`
- `book/src/advanced_blobs.md`
- CLI help text for `--checkpoint-blobs` flag

### 🎯 Verdict

The change is **sound and safe**. The core functionality should work correctly given #8417's foundation. Test coverage for the None blobs path would provide additional confidence and prevent future regressions.

**Review by jimmygchen — COMMENTED** [review]
## Code Review

### ✅ Strengths

1. **Proper prerequisite**: Correctly builds on #8417 which enables backfill to re-import the checkpoint sync block
2. **Safety compliance**: No `.unwrap()`, `.expect()`, or unsafe array indexing
3. **Logic simplification**: Removes complex `is_before_fulu` check, making code more straightforward
4. **Clear documentation**: Updated comment explains the change well

### ⚠️ Recommendations

#### 1. Missing test coverage
The existing test at `beacon_node/beacon_chain/tests/store_tests.rs:2874` still provides blobs via `wss_blobs_opt.clone()`. There's no test covering the new code path where blobs are NOT provided and backfill must fetch them.

**Recommendation**: Add a test case that passes `None` for `anchor_blobs_bytes` to verify backfill correctly handles this scenario.

#### 2. Consider adding logging
When a block has blobs but the user doesn't provide them, there's no log message indicating that backfill will handle fetching them. This could aid debugging.

**Suggestion**: Add a log when blobs are omitted:
```rust
} else {
    if anchor_block.message().body().has_blobs() {
        info!("Checkpoint block has blobs/columns which will be fetched via backfill");
    }
    None
};
```

#### 3. Documentation updates
Since this changes user-facing behavior (users no longer need `--checkpoint-blobs`), verify these don't need updates:
- `book/src/advanced_checkpoint_sync.md`
- `book/src/advanced_blobs.md`
- CLI help text for `--checkpoint-blobs` flag

### 🎯 Verdict

The change is **sound and safe**. The core functionality should work correctly given #8417's foundation. Test coverage for the None blobs path would provide additional confidence and prevent future regressions.

**Review by jimmygchen — COMMENTED** [review]
**Suggestion for beacon_node/client/src/builder.rs around line 357:**

Consider adding a log message when blobs are not provided but the block has them:

```rust
let anchor_blobs = if let Some(anchor_blobs_bytes) = anchor_blobs_bytes {
    let max_blobs_len = spec.max_blobs_per_block(anchor_block.epoch()) as usize;
    Some(
        BlobSidecarList::from_ssz_bytes(&anchor_blobs_bytes, max_blobs_len)
            .map_err(|e| format!("Unable to parse weak subj blobs SSZ: {e:?}"))?,
    )
} else {
    if anchor_block.message().body().has_blobs() {
        info!("Checkpoint block has blobs/columns which will be fetched via backfill");
    }
    None
};
```

This would help users understand what's happening when they omit `--checkpoint-blobs` and aid debugging if backfill encounters issues.

File: beacon_node/client/src/builder.rs [hunk 1]
```diff
@@ -354,15 +354,10 @@ where
...
 
-                // `BlobSidecar` is no longer used from Fulu onwards (superseded by `DataColumnSidecar`),
-                // which will be fetched via rpc instead (unimplemented).
-                let is_before_fulu = !spec
-                    .fork_name_at_slot::<E>(anchor_block.slot())
-                    .fulu_enabled();
-                let anchor_blobs = if is_before_fulu && anchor_block.message().body().has_blobs() {
+                // Providing blobs is optional now and not providing them is recommended.
+                // Backfill can handle downloading the blobs or columns for the checkpoint block.
+                let anchor_blobs = if let Some(anchor_blobs_bytes) = anchor_blobs_bytes {
```
Comment (jimmygchen) [inline beacon_node/client/src/builder.rs:359]: Sorry for the verbose reviews above. Here are the key points concisely:

1. Missing test for None blobs path - existing test at `store_tests.rs:2874` still provides blobs
2. Consider logging when blobs omitted but block has them (aids debugging)
3. Check if these docs need updating: `book/src/advanced_checkpoint_sync.md`, `advanced_blobs.md`, CLI help for `--checkpoint-blobs`

Otherwise looks good - builds correctly on #8417.

Comment (jimmygchen) [reply beacon_node/client/src/builder.rs:359]: Test added here https://github.com/sigp/lighthouse/pull/8473/files

Comment (michaelsproul) [reply beacon_node/client/src/builder.rs:359]: Thanks!

**Review by eserilev — APPROVED** [review]
LGTM!
